rest-api/dao/questionnaire_response_dao.py

```diff
@@ -228,6 +229,15 @@ def _update_participant_summary(
         participant_summary.race = race
         something_changed = True
 
+    # if primary language provided in the response, set the new value.
+    resource_json = json.loads(questionnaire_response.resource)
```

Comment:
Rather than re-parsing resource_json here, we should have _update_participant_summary take either the JSON or the language as a parameter.